x-pack/plugins/infra/public/plugin.ts

```diff
@@ -50,6 +51,29 @@ export class Plugin implements InfraClientPluginClass {
       });
     }
 
+    /** !! Need to be kept in sync with the deepLinks in x-pack/plugins/infra/public/plugin.ts */
+    pluginsSetup.observability.navigation.registerSections(
+      of([
```

Comment:
This unconditionally registers the apps regardless of the space and role settings. We probably need to check the `show` capabilities as in

![image](https://user-images.githubusercontent.com/973741/120819369-cc91d980-c553-11eb-94ba-0e55fd4f025b.png)

Reply:
Oh wow, yep 🙈 

Of course I would remember to add the comment back left by the platform team, but forget the capability check.

Also related: I think we should make the Observability plugin required. It's currently optional. I would either need to move this registration into our `if`, or we make it required. Now that navigation is routed through the Observability plugin I think it's fair to require it.

Reply:
> I think we should make the Observability plugin required.

Agreed. The impact on the user should be minimal since there is no separate permission for the observability plugin to take away and disabling plugins via the `yaml` is probably going to go away soon.